Modification to make backtransformations easier

snjnz opened the pull request:
Changes to multipleComp, summary2way and estimateConstrasts to accept a
function `FUN` for the purposes of backtransformation, although in most
cases the functions can be wrapped with `exp()` this also transforms
p-values which can confound students and usually results in interesting
code solutions to work around the issue.

The main goal here is (without breaking existing code/examples - hence defaulting to identity) exposing a method for users/students to backtransform the inner parts of the tukey intervals without having to know matrix sub-setting, or as is more common, getting confused with p-values going erratic.

Commit:
Modification to make backtransformations easier

Changes to multipleComp, summary2way and estimateConstrasts to accept a
function `FUN` for the purposes of backtransformation, although in most
cases the functions can be wrapped with `exp()` this also transforms
p-values which can confound students and usually results in interesting
code solutions to work around the issue.

jmcurran reviewed (changes requested):
Nigel - I am generally in favour of this. However, I the back-transformation appears to be applied to the printed output, but not to the returned object. Is/was that your intention?

```
> res = summary2way(fit, page = "nointeraction", FUN = exp)
  Tukey multiple comparisons of means
    95% family-wise confidence level

Fit: aov(formula = fit)

$Breed
                         diff       lwr       upr     p adj
canadian-ayrshire   1.0948168 1.0134408 1.1827271 0.0130682
guernesy-ayrshire   1.2160909 1.1257007 1.3137391 0.0000000
holst.fres-ayrshire 0.9035802 0.8364185 0.9761348 0.0038416
jersey-ayrshire     1.2983817 1.2018750 1.4026376 0.0000000
guernesy-canadian   1.1107711 1.0282092 1.1999624 0.0024730
holst.fres-canadian 0.8253255 0.7639803 0.8915965 0.0000000
jersey-canadian     1.1859351 1.0977863 1.2811618 0.0000002
holst.fres-guernesy 0.7430203 0.6877928 0.8026825 0.0000000
jersey-guernesy     1.0676683 0.9883102 1.1533986 0.1362342
jersey-holst.fres   1.4369302 1.3301254 1.5523110 0.0000000

$Age
                diff       lwr      upr     p adj
mature-2yo 0.9807275 0.9471187 1.015529 0.2706475

> res$results$comparisons$Breed
                           diff         lwr         upr        p adj
canadian-ayrshire    0.09058708  0.01335127  0.16782289 1.306815e-02
guernesy-ayrshire    0.19564150  0.11840570  0.27287731 3.507400e-09
holst.fres-ayrshire -0.10139038 -0.17862619 -0.02415457 3.841631e-03
jersey-ayrshire      0.26111862  0.18388281  0.33835443 4.836335e-10
guernesy-canadian    0.10505442  0.02781861  0.18229023 2.472983e-03
holst.fres-canadian -0.19197746 -0.26921327 -0.11474165 6.092863e-09
jersey-canadian      0.17053154  0.09329573  0.24776735 1.936339e-07
holst.fres-guernesy -0.29703188 -0.37426769 -0.21979607 4.835826e-10
jersey-guernesy      0.06547712 -0.01175869  0.14271293 1.362342e-01
jersey-holst.fres    0.36250900  0.28527319  0.43974481 4.835520e-10
```

Commit:
Update summary2way to also update returned list

Unaltered results will be retained in results.identity

This should be backwards compatible because FUN will default to identity

snjnz commented:
Yeah, nice spotting, it should've also backtransformed the means/effects tables (I'd suggest renaming the 'means' table but since we can't always tell what backtransform will be applied, we'll have issues there...

I've changed it to return a `results` and `results.identity` object to the list - `results.identity` as the untransformed data, while `results` taking the backtransformed data to match what is printed.  I'll need to update the docstrings to note the new output format...

Commit:
Fix "Error in out[[1]]: subscript out of bounds"

Calling summary2way(fit, page="interaction") where `fit` is not an
interaction model will result in the subscript out of bounds error. We
test for if an interaction - 2nd order - term is present, so use that to
present a user-friendly error.

snjnz commented:
I've just snuck in another fix unrelated to the main set of fixes, basically if `page="interaction"` but the model doesn't have a 2nd order term, there is a rather unfriendly error as it tries to access part of the `out` list that doesn't exist. Quick low-risk touch up to throw a `stop` with a friendlier error.
